### PR TITLE
les: wg data race fixed

### LINKED ISSUE #15364: Data race in les.Stop()
#### System information

Geth version: 1.7.2
OS & Version: Windows/Linux/OSX
Commit hash : 6d6a5a93370371a33fb815d7ae47b60c7021c86a

#### Expected behaviour
No data races in the code.

#### Actual behaviour
Data race caused unsave operation with wait group: any wg.Add() should be after wg.Wait().

#### Steps to reproduce the behaviour
I've found this data race while running test of Status-IM project: `github.com/status-im/status-go/e2e/node_test.(*ManagerTestSuite).TestNodeStartCrash()`

#### Backtrace

````
==================
WARNING: DATA RACE
Write at 0x00c4200fa46c by goroutine 106:
  internal/race.Write()
      /usr/local/go/src/internal/race/race.go:41 +0x38
  sync.(*WaitGroup).Wait()
      /usr/local/go/src/sync/waitgroup.go:129 +0xf4
  github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/les.(*ProtocolManager).Stop()
      /home/eugene/go/src/github.com/status-im/status-go/build/_workspace/project/src/github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/les/handler.go:246 +0x114
  github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/les.(*LightEthereum).Stop()
      /home/eugene/go/src/github.com/status-im/status-go/build/_workspace/project/src/github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/les/backend.go:222 +0xb3
  github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/node.(*Node).Stop()
      /home/eugene/go/src/github.com/status-im/status-go/build/_workspace/project/src/github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/node/node.go:488 +0x337
...

Previous read at 0x00c4200fa46c by goroutine 35:
  internal/race.Read()
      /usr/local/go/src/internal/race/race.go:37 +0x38
  sync.(*WaitGroup).Add()
      /usr/local/go/src/sync/waitgroup.go:71 +0x16f
  github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/les.(*lightFetcher).syncLoop()
      /home/eugene/go/src/github.com/status-im/status-go/build/_workspace/project/src/github.com/status-im/status-go/vendor/github.com/ethereum/go-ethereum/les/fetcher.go:126 +0x95
````

### PR DESCRIPTION
**Pull request by JekaMas**
Fixes #15364

### CODE REVIEW

**Comment by zsfelfoldi** [comment]
Thanks for catching it! There is a better way to fix this though. wg.Add should not happen inside the goroutine but before starting it. Then it is guaranteed that it runs in the init phase, before starting the entire LightEthereum backend and the additional semaphore is not needed at all.

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: 98b63fbdae37d4137f8deb00ebb30d0fb47005c2
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

**Comment by JekaMas** [comment]
@zsfelfoldi Thanks for advice! I've make requested changes.

File: les/handler.go [hunk 1]
```diff
@@ -242,7 +243,7 @@ func (pm *ProtocolManager) Stop() {
 	// will exit when they try to register.
 	pm.peers.Close()
 
-	// Wait for any process action
+	// Wait for any process action. Wait should be executed after all pm.wg.Add()
```
Comment (fjl) [inline les/handler.go]: Please remove this comment.

File: les/handler.go [hunk 2]
```diff
@@ -115,6 +115,7 @@ type ProtocolManager struct {
 
 	// channels for fetcher, syncer, txsyncLoop
 	newPeerCh   chan *peer
+	isClosed	*int32
```
Comment (fjl) [inline les/handler.go]: This is left over from your previous change, please remove.

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: 596c13b55d36197a5c7887f32096cfc202d85f61
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

File: les/handler.go (hunk 1 above)
Comment (JekaMas) [reply les/handler.go]: Thanks. This and upper changes has been done.

**Pull request merged**